### ISSUE TITLE
lint: add suspicious op priority-related issues checker

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#568]
Detects suspicious code that may be incorrect due to operator priority.

**Before:**
```go
return enabled && op == A || op == B
```
**After:**
```go
return enabled && (op == A || op == B)
```
----

Quite hard (impossible?) to do without false positives.
May be a good fit for gocritic.

**Comment by dominikh** [comment]
Do note that this is less of a problem in Go than in other languages, for two reasons:

1. gofmt groups binary ops by precedence, so from reading the code, the order of operations tends to be obvious
2. the biggest troublemaker in C, the combination of binary bitwise ops and bit shifts, behaves correctly in Go due to different precedence for these operators.
